internal/services/securitycenter/security_center_storage_defender_resource.go

```diff
@@ -0,0 +1,314 @@
+	return sdk.ResourceFunc{
+		Timeout: 10 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var plan StorageDefenderModel
+			if err := metadata.Decode(&plan); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.SecurityCenter.DefenderForStorageClient
+
+			id, err := commonids.ParseScopeID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			model := resp.Model
+			if model == nil {
+				return fmt.Errorf("retrieving %s: model was nil", *id)
+			}
+
+			prop := model.Properties
+			if prop == nil {
+				return fmt.Errorf("retrieving %s: properties was nil", *id)
+			}
+
+			if metadata.ResourceData.HasChange("enabled") {
+				prop.IsEnabled = pointer.To(plan.Enabled)
+			}
+
+			if metadata.ResourceData.HasChange("override_subscription_settings_enabled") {
+				prop.OverrideSubscriptionLevelSettings = pointer.To(plan.OverrideSubscriptionSettings)
+			}
+
+			if prop.MalwareScanning == nil {
+				prop.MalwareScanning = &defenderforstorage.MalwareScanningProperties{}
+			}
+
+			if prop.MalwareScanning.OnUpload == nil {
+				prop.MalwareScanning.OnUpload = &defenderforstorage.OnUploadProperties{}
+			}
+
+			if metadata.ResourceData.HasChange("malware_scanning_on_upload_enabled") {
+				prop.MalwareScanning.OnUpload.IsEnabled = pointer.To(plan.MalwareScanningOnUploadEnabled)
+			}
+
+			if metadata.ResourceData.HasChange("malware_scanning_on_upload_cap_gb_per_month") {
+				prop.MalwareScanning.OnUpload.CapGBPerMonth = pointer.To(plan.MalwareScanningOnUploadCapPerMon)
+			}
+
+			if prop.SensitiveDataDiscovery == nil {
+				prop.SensitiveDataDiscovery = &defenderforstorage.SensitiveDataDiscoveryProperties{}
+			}
+
+			if metadata.ResourceData.HasChange("sensitive_data_discovery_enabled") {
+				prop.SensitiveDataDiscovery.IsEnabled = pointer.To(plan.SensitiveDataDiscoveryEnabled)
+			}
+
+			input := defenderforstorage.DefenderForStorageSetting{
+				Properties: prop,
+			}
+
+			_, err = client.Create(ctx, *id, input)
+			if err != nil {
+				return fmt.Errorf("updating: %+v", err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (s StorageDefenderResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.SecurityCenter.DefenderForStorageClient
+
+			id, err := commonids.ParseScopeID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("parsing %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %+v", err)
+			}
+
+			state := StorageDefenderModel{
+				StorageAccountId: id.ID(),
+			}
```

Comment:
Since this is a scoped ID, we need to parse the scope part of `id` as a storage account ID
```suggestion
			storageAccountId, err := commonids.ParseStorageAccountID(id.Scope)
			if err != nil {
				return err
			}
			state := StorageDefenderModel{
				StorageAccountId: storageAccountId.ID(),
			}
```